Fix part of #11776. Adding/Updating 'Args' and 'Return' in docstrings

deadex-ng opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #11776 .
2. This PR does the following: Added/updated args, return and raises in docstrings

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng committed:
fix pylint issues

deadex-ng commented:
@gp201 PTAL

gp201 reviewed (commented):
@deadex-ng Is it possible to explain what the function aims to do rather than a generic `Handles GET requests` doc string?

deadex-ng commented:
> @deadex-ng Is it possible to explain what the function aims to do rather than a generic `Handles GET requests` doc string?

I think that is quite complex.

deadex-ng committed:
remove generic docstrings

deadex-ng committed:
fix linter issues

deadex-ng commented:
@gp201 I have removed the generic doc string in [admin.py](https://github.com/oppia/oppia/pull/18216/files#diff-981a56d20653e10899f2301ebcb06864bda37939c907e36965e2840ae1c36cbb) and updated it.

deadex-ng commented:
@gp201 PTAL

deadex-ng committed:
remove generic docstrings

gp201 commented on core/controllers/admin.py:
```diff
@@ -1889,6 +1925,9 @@ class UpdateBlogPostHandler(
 
     @acl_decorators.can_access_admin_page
     def put(self) -> None:
+        """Handles PUT requests to update the author and published date of
+        a blog post.
```
Maybe, "Updates the author and published date of a blog post". you don't have to add "Handles PUT requests"

deadex-ng replied:
Alright.

gp201 commented on core/controllers/admin.py:
```diff
@@ -1783,6 +1812,10 @@ class VerifyUserModelsDeletedHandler(
 
     @acl_decorators.can_access_admin_page
     def get(self) -> None:
+        """Checks if a user with a specific user_id has been deleted and
+        returns a JSON response indicating whether there are related models
+        or not.
+        """
```
Can you try to rephrase it to be a bit shorter?

deadex-ng replied:
"Checks if a user with a specific user_id has been deleted and
 if there are related models or not."

@gp201  Does this work?

gp201 reviewed (changes requested):
Thanks, @deadex-ng ! I left some comments. PTAL

deadex-ng committed:
updated args, return and raises in docstring

deadex-ng commented:
@gp201 I have made the changes you suggested. PTAL

gp201 reviewed (changes requested):
@deadex-ng nice work! Can  you make similar changes to the remaining files. Note: You don't have to `Handles ___ requests` to the beginning of every  docstring

deadex-ng committed:
update generic docstrings

deadex-ng committed:
remove generic docstrings

deadex-ng commented:
@gp201 @seanlip @chris7716 PTAL

seanlip commented on core/controllers/classroom.py:
```diff
@@ -294,7 +310,11 @@ class ClassroomUrlFragmentHandler(
     @acl_decorators.can_access_classroom_admin_page
     def get(self, classroom_url_fragment: str) -> None:
-        """Get request to check whether a classroom with given exists."""
+        """Get request to check whether a classroom with given exists.
```
Could you rewrite this description to be clearer? I'm not sure what it means.

deadex-ng replied:
I have edited this.

seanlip commented on core/controllers/classroom.py:
```diff
@@ -319,6 +339,7 @@ class ClassroomIdHandler(
 
     @acl_decorators.open_access
     def get(self, classroom_url_fragment: str) -> None:
+        """Retrieves the classroom ID."""
```
Missing Args section.

deadex-ng replied:
I have updated the docstrings.

seanlip commented on core/controllers/learner_goals.py:
```diff
@@ -65,6 +65,7 @@ class LearnerGoalsHandler(
     # body to handle the different activity cases properly.
     @acl_decorators.can_access_learner_dashboard
     def post(self, activity_type: str, topic_id: str) -> None:  # pylint: disable=unused-argument
+        """Adds a topic to the learner's learning goals."""
```
Missing Args section, here and below.

deadex-ng replied:
I have updated the docstrings.

seanlip commented on core/controllers/practice_sessions.py:
```diff
@@ -113,6 +113,7 @@ class PracticeSessionsPageDataHandler(
 
     @acl_decorators.can_access_topic_viewer_page
     def get(self, topic_name: str) -> None:
+        """Retrieves information about a topic."""
```
Missing Args section.

deadex-ng replied:
I have updated the docstrings.

seanlip commented on core/controllers/release_coordinator.py:
```diff
@@ -38,6 +38,7 @@ class MemoryCacheHandler(
 
     @acl_decorators.can_manage_memcache
     def get(self) -> None:
+        """Retrieve statistics about the memory cache."""
```
Retrieve --> Retrieves 

(for consistency)

deadex-ng replied:
I have updated the docstrings.

seanlip commented on core/controllers/skill_editor.py:
```diff
@@ -281,7 +281,7 @@ def put(self, skill_id: str) -> None:
     @acl_decorators.can_delete_skill
     def delete(self, skill_id: str) -> None:
-        """Handles Delete requests."""
+        """Deletes a skill."""
```
Missing Args.

deadex-ng replied:
I have updated the docstrings.

seanlip commented on core/controllers/story_editor.py:
```diff
@@ -188,7 +188,7 @@ def put(self, story_id: str) -> None:
     @acl_decorators.can_delete_story
     def delete(self, story_id: str) -> None:
-        """Handles Delete requests."""
+        """Deletes a story."""
```
Missing Args. Could you please proofread the PR for these?

deadex-ng replied:
I have updated the docstrings.

seanlip commented on core/controllers/story_viewer.py:
```diff
@@ -217,6 +222,15 @@ def get(self, story_id: str, node_id: str) -> None:
+            node_id: str. The node ID.
+
+        Returns:
+            json. A json object.
```
This seems wrong.

deadex-ng replied:
The removing Returns gives the following lint check error.
```
************* Module core.controllers.story_viewer
core/controllers/story_viewer.py:228:4: W9011: Missing return documentation 
core/controllers/story_viewer.py:228:4: W9012: Missing return type documentation 

---------------------------
Linter Checks Failed.
---------------------------

```

seanlip replied:
Hm, if you look at the body of the code, you'll see that there's a "return" at the end. But this is different from all the other handlers, so I think it's an error. Can you remove the "return" before self.return_json at the bottom so that it matches the other handlers?

deadex-ng replied:
I removed it and it works now.

seanlip replied:
OK, thanks. Feel free to request a review once you've made the changes and replied to all the other comments as well.

seanlip commented on core/controllers/topic_editor.py:
```diff
@@ -833,6 +875,7 @@ class TopicIdToTopicNameHandler(
 
     @acl_decorators.can_access_classroom_admin_page
     def get(self) -> None:
+        """Access a classroom admin page."""
```
Access --> Accesses (let's keep tenses consistent)

deadex-ng replied:
I have updated the docstrings.

seanlip commented on core/controllers/topic_viewer.py:
```diff
@@ -61,7 +61,11 @@ class TopicPageDataHandler(base.BaseHandler[Dict[str, str], Dict[str, str]]):
     @acl_decorators.can_access_topic_viewer_page
     def get(self, topic_name: str) -> None:
-        """Handles GET requests."""
+        """Access a topic viewer page.
```
Access --> Accesses (let's keep tenses consistent)

deadex-ng replied:
I have updated the docstrings.

seanlip reviewed (changes requested):
Thanks, @deadex-ng! Reviewed codeowner files, PTAL.

seanlip reviewed (changes requested):
Thanks, @deadex-ng! Reviewed codeowner files, PTAL.

seanlip reviewed (changes requested):
Thanks, @deadex-ng! Reviewed codeowner files, PTAL.

seanlip reviewed (changes requested):
Thanks, @deadex-ng! Reviewed codeowner files, PTAL.

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix linter issues

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix linter issues

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
fix linter issues

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng commented:
@seanlip PTAL

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -266,7 +266,7 @@ def get(self) -> None:
     @acl_decorators.can_access_admin_page
     def post(self) -> None:
-        """Handles POST requests."""
+        """Performs a series of actions based on the action parameter."""
```
```suggestion
        """Performs a series of actions based on the action parameter on the admin page."""
```

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1178,6 +1178,9 @@ class AdminRoleHandler(
     def get(self) -> None:
+        """Handles GET requests to retrieve information about users based on
+        different filter criteria.
+        """
```
I think you can avoid adding `Handles GET/PUT/POST.. requests` at the beginning. Also, it needs to be rephrased. This is part of the `AdminRoleHandler` and the docstring does not mention anything about roles.
Additionally, missing `Raises` part from the docstring.

deadex-ng replied:
@lkbhitesh07  Is this correct? 
Especially the part about `AdminRoleHandler`
```
        """Retrieves information about users based on different filter
        criteria. It expects a request of type
        AdminRoleHandlerNormalizedGetRequestDict

        Raises:
            Exception. The role must be provided when the filter criterion
                is 'role'.
            Exception. The username must be provided when the filter
                criterion is 'username'.
            InvalidInputException. User with given username does not exist.
        """
```

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1244,6 +1247,7 @@ def get(self) -> None:
 
     @acl_decorators.can_access_admin_page
     def put(self) -> None:
+        """Handles PUT requests to assign a role to a user."""
```
Avoid using the `Handles ...` part. Add the `Raises` part to the docstring. You can rephrase it to be something like `Updates the roles for the user.`

deadex-ng replied:
I have updated the docstring to fix this.

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1265,6 +1269,7 @@ def put(self) -> None:
 
     @acl_decorators.can_access_admin_page
     def delete(self) -> None:
+        """Handles DELETE requests to remove a role from a user."""
```
ditto

deadex-ng replied:
I have updated the docstring to fix this.

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1331,6 +1336,9 @@ class TopicManagerRoleHandler(
     def put(self) -> None:
+        """Assigns or deassigns the feconf.ROLE_ID_TOPIC_MANAGER role
+        for a user in the context of a specific topic.
+        """
```
```suggestion
        """Adds or removes the topic-manager role for a user
        in the context of a specific topic.
        """
```

Also, missing `Raises`.

deadex-ng replied:
I have updated the docstring to fix this.

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1416,6 +1424,7 @@ class BannedUsersHandler(
 
     @acl_decorators.can_access_admin_page
     def put(self) -> None:
+        """Handles PUT requests to mark a user as banned by their username."""
```
ditto

deadex-ng replied:
I have updated the docstring to fix this.

lkbhitesh07 reviewed (changes requested):
Hey @deadex-ng, thanks for the changes. I have added some comments.
The general comment for the `admin.py` file would be to avoid the usage of `Handles __ requests` part and to add the `Raises` part where missing. Wherever I mention "ditto" I mean these 2 changes mainly.
Please assign me again after the changes.

Thanks

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix linter issues

seanlip commented on core/controllers/skill_editor.py:
```diff
@@ -281,7 +306,14 @@ def put(self, skill_id: str) -> None:
+            skill_id: str. The skill ID.
+
+        Raises:
+            InvalidInputException. Please delete all skill questions.
```
The part after the InvalidInputException should explain what condition causes the exception to be raised, not show the error message. See e.g. what you did for SkillDataHandler.get() below -- that one is correct.

deadex-ng replied:
In instances where the function is taking multiple parameters, I have summarized it to 
`The input provided is not valid.`. Is that okay?
```
    @acl_decorators.can_edit_story
    def put(self, story_id: str) -> None:
        """Updates properties of the given story.

        Args:
            story_id: str. The story ID.

        Raises:
            InvalidInputException. The input provided is not valid.
        """
        assert self.user_id is not None
        assert self.normalized_payload is not None
        story = story_fetchers.get_story_by_id(story_id, strict=True)
        version = self.normalized_payload['version']
        commit_message = self.normalized_payload['commit_message']
        change_dicts = self.normalized_payload['change_dicts']
        self._require_valid_version(version, story.version)

        try:
            # Update the Story and its corresponding TopicSummary.
            topic_services.update_story_and_topic_summary(
                self.user_id, story_id, change_dicts, commit_message,
                story.corresponding_topic_id)
        except utils.ValidationError as e:
            raise self.InvalidInputException(e)

        story_dict = story_fetchers.get_story_by_id(story_id).to_dict()

        self.values.update({
            'story': story_dict
        })

        self.render_json(self.values)
```

seanlip replied:
Hm, I think that's OK given that it's a ValidationError causing this. Thanks for checking!

seanlip commented on core/controllers/story_editor.py:
```diff
@@ -161,7 +175,14 @@ def get(self, story_id: str) -> None:
+            story_id: str. The story ID.
+
+        Raises:
+            InvalidInputException. Error class for invalid input.
```
See above on how to document this properly.

deadex-ng replied:
I have made the changes.

seanlip commented on core/controllers/story_editor.py:
```diff
@@ -228,7 +253,11 @@ class StoryPublishHandler(
     @acl_decorators.can_edit_story
     def put(self, story_id: str) -> None:
-        """Published/unpublished given story."""
+        """Published/unpublished given story.
```
Please keep the tenses consistent with other handlers.

deadex-ng replied:
I have changed the tenses.

seanlip commented on core/controllers/story_viewer.py:
```diff
@@ -171,7 +175,12 @@ def _record_node_completion(
     @acl_decorators.can_access_story_viewer_page_as_logged_in_user
     def get(self, story_id: str, node_id: str) -> None:
-        """Handles GET requests."""
+        """Views a specific node within a story.
```
In other handlers, the implicit subject for the top-line docstring is the handler, i.e. "This handler records the completion of ...".

This one seems different though (the handler isn't viewing anything, it's providing information to the user). Please check your docstrings to make sure they are written "from the handler's perspective".

deadex-ng replied:
Yes, you're right. I have made the changes.

seanlip commented on core/controllers/topic_editor.py:
```diff
@@ -585,7 +613,14 @@ class TopicRightsHandler(
+            topic_id: str. The ID of the topic.
+
+        Raises:
+            InvalidInputException. Expected a valid topic id to be provided.
```
InvalidInputException: The topic ID provided is not valid.

deadex-ng replied:
I have made the changes.

seanlip commented on core/controllers/topics_and_skills_dashboard.py:
```diff
@@ -307,7 +310,7 @@ class SkillsDashboardPageDataHandler(
     @acl_decorators.can_access_topics_and_skills_dashboard
     def post(self) -> None:
-        """Handles POST requests."""
+        """filters and fetches skill summaries."""
```
Start docstrings with a capital letter.

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng commented:
@seanlip PTAL

seanlip reviewed (approved):
LGTM for codeowner files. Thanks @deadex-ng!

gp201 commented on core/tests/test_utils.py:
```diff
@@ -286,15 +286,23 @@ def check_image_png_or_webp(image_string: str) -> bool:
+    """Get all module names in storage.
+
+    Yields:
+        Iterator. The Iterator object.
```
Please take a look at how iterator values are defined in other files.
example:
https://github.com/oppia/oppia/blob/8d821797c8b941347bf0b33d110d0923da96209e/core/jobs/jobs_manager.py#L257

deadex-ng replied:
I have updated the docstring.

gp201 commented on core/tests/test_utils.py:
```diff
@@ -2818,7 +2840,15 @@ def get_response_without_checking_for_errors(
+            expect_errors: bool. Whether errors are expected.
+
+        Returns:
+            Any. Python object.
```
Please be more specific. The code seems to be returning some sort of json

deadex-ng replied:
I have updated the docstring.

gp201 commented on core/tests/test_utils.py:
```diff
@@ -2842,7 +2872,17 @@ def get_json(
+            headers: Dict[str, str]. The headers used in the request.
+
+        Returns:
+            Any. A python object.
```
*Ditto

deadex-ng replied:
I have updated the docstring.

gp201 reviewed (changes requested):
Thanks, @deadex-ng ! I left some comments. PTAL

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix linter issues

deadex-ng commented:
@gp201 PTAL

JayVivarekar commented on core/controllers/access_validators.py:
```diff
@@ -63,6 +63,7 @@ class ClassroomAccessValidationHandler(
 
     @acl_decorators.open_access
     def get(self) -> None:
+        """Retrieves information about a classroom."""
```
Raises missing

deadex-ng replied:
I have updated the docstring.

JayVivarekar replied:
Not updated

JayVivarekar commented on core/controllers/access_validators.py:
```diff
@@ -110,7 +112,11 @@ class ProfileExistsValidationHandler(
     @acl_decorators.open_access
     def get(self, username: str) -> None:
-        """Validates access to profile page."""
+        """Validates access to profile page.
```
Missing 'Raises'

deadex-ng replied:
I have updated the docstring.

JayVivarekar commented on core/controllers/access_validators.py:
```diff
@@ -161,7 +167,11 @@ class ViewLearnerGroupPageAccessValidationHandler(
+
+        Args:
+            learner_group_id: str. The learner group ID.
+        """
```
Ditto as above

deadex-ng replied:
I have updated the docstring.

JayVivarekar commented on core/controllers/access_validators.py:
```diff
@@ -221,6 +231,7 @@ class BlogPostPageAccessValidationHandler(
 
     @acl_decorators.open_access
     def get(self) -> None:
+        """Retrieves information about a blog post."""
```
Ditto as above

JayVivarekar replied:
Not updated

JayVivarekar commented on core/controllers/access_validators.py:
```diff
@@ -255,6 +266,11 @@ class BlogAuthorProfilePageAccessValidationHandler(
+
+        Args:
+            author_username: str. The author username.
+        """
```
Missing raises

deadex-ng replied:
I have updated the docstring.

JayVivarekar commented on core/controllers/admin.py:
```diff
@@ -1331,6 +1384,12 @@ class TopicManagerRoleHandler(
 
     @acl_decorators.can_access_admin_page
     def put(self) -> None:
+        """Adss or removes the topic-manager role for a user in the context
```
```suggestion
        """Adds or removes the topic-manager role for a user in the context
```

deadex-ng replied:
I have updated the docstring.

JayVivarekar replied:
Not done

JayVivarekar commented on core/controllers/blog_dashboard.py:
```diff
@@ -325,7 +333,11 @@ def put(self, blog_post_id: str) -> None:
     @acl_decorators.can_edit_blog_post
     def post(self, blog_post_id: str) -> None:
-        """Stores thumbnail of the blog post in the datastore."""
+        """Stores thumbnail of the blog post in the datastore.
```
Missing 'Raises'.

deadex-ng replied:
I have updated the docstring.

JayVivarekar replied:
Not done

JayVivarekar commented on core/controllers/blog_dashboard.py:
```diff
@@ -349,7 +361,11 @@ def post(self, blog_post_id: str) -> None:
     @acl_decorators.can_delete_blog_post
     def delete(self, blog_post_id: str) -> None:
-        """Handles Delete requests."""
+        """Deletes blog post.
```
```suggestion
        """Deletes a blog post.
```

JayVivarekar replied:
Not done

JayVivarekar commented on core/controllers/blog_homepage.py:
```diff
@@ -249,7 +249,11 @@ class BlogPostDataHandler(
     @acl_decorators.open_access
     def get(self, blog_post_url: str) -> None:
-        """Handles GET requests."""
+        """Retrieves a specific blog post and its related recommendations.
```
Missing "Raises"

deadex-ng replied:
I have updated the docstring.

JayVivarekar replied:
Not done

JayVivarekar commented on core/controllers/classroom.py:
```diff
@@ -245,7 +249,11 @@ class ClassroomHandler(
     @acl_decorators.open_access
     def get(self, classroom_id: str) -> None:
-        """Handles GET requests."""
+        """Retrieves the classroom details.
```
Missing 'Raises'

deadex-ng replied:
I have updated the docstring.

JayVivarekar replied:
Not done

JayVivarekar commented on core/controllers/classroom.py:
```diff
@@ -259,7 +267,11 @@ def get(self, classroom_id: str) -> None:
     @acl_decorators.can_access_classroom_admin_page
     def put(self, classroom_id: str) -> None:
-        """Updates properties of a given classroom."""
+        """Updates properties of a given classroom.
```
Ditto as above

deadex-ng replied:
I have updated the docstring.

JayVivarekar reviewed (changes requested):
Thanks @deadex-ng! Took a pass.

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng committed:
fix linter issues

deadex-ng commented:
@JayVivarekar PTAL

JayVivarekar reviewed (changes requested):
Thanks @deadex-ng ! Some of the comments were not addressed. PTAL

deadex-ng commented:
> Thanks @deadex-ng ! Some of the comments were not addressed. PTAL

Thanks @JayVivarekar! I have addressed all your comments. I have added 'Args'  and 'Raises' for each method where necessary.
Could you please be more specific on where those are missing?

@JayVivarekar PTAL

chris7716 commented on core/controllers/contributor_dashboard.py:
```diff
@@ -315,7 +315,7 @@ class ReviewableOpportunitiesHandler(
     @acl_decorators.open_access
     def get(self) -> None:
-        """Handles GET requests."""
+        """Fetches exploration opportunities."""
```
```suggestion
        """Fetches reviewable translation suggestions."""
```

deadex-ng replied:
I have updated the docstring with the suggested change.

chris7716 commented on core/controllers/contributor_dashboard.py:
```diff
@@ -1006,7 +1006,12 @@ class ContributorCertificateHandler(
         self, username: str, suggestion_type: str
     ) -> None:
-        """Handles GET requests."""
+        """Generates contributor certificates.
```
```suggestion
        """Generates data for contributor certificates.
```

deadex-ng replied:
I have updated the docstring with the suggested change.

chris7716 commented on core/controllers/contributor_dashboard.py:
```diff
@@ -1047,7 +1052,11 @@ class ContributorAllStatsSummariesHandler(
     @acl_decorators.can_fetch_all_contributor_dashboard_stats
     def get(self, username: str) -> None:
-        """Handles GET requests."""
+        """Fetches contributor dashboard statistics.
```
```suggestion
        """Fetches stats for given contributor.
```

deadex-ng replied:
I have updated the docstring with the suggested change.

chris7716 reviewed (changes requested):
@deadex-ng Thanks! Took a pass.

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng commented:
@chris7716 PTAL

chris7716 reviewed (approved):
LGTM for codeowner files. Thanks @deadex-ng

gp201 reviewed (approved):
LGTM for codeowner files

JayVivarekar reviewed (commented):
Hi @deadex-ng. I have pointed out the places where changes were not made with "Not done". For example consider this file (access_validators.py).
https://github.com/oppia/oppia/pull/18216/files/50a7f1def5cf378ddfd558a05c3c6fb8e4855731#diff-1d44d96af9e450b5ca994420de1d64c5fcc9d603666cb8d8321a16e8908e3c74

The `get` method can raise a `PageNotFoundException` but the docstring hasn't been updated. Similarly for other files I have mentioned. I hope that answers your query.

deadex-ng commented on core/controllers/access_validators.py:
```diff
@@ -63,6 +63,11 @@ class ClassroomAccessValidationHandler(
+        Raises:
+            PageNotFoundException. The classroom cannot be found.
+        """
         assert self.normalized_request is not None
```
@JayVivarekar I think you mean this part [here](https://github.com/oppia/oppia/pull/18216#pullrequestreview-1459287158)

deadex-ng commented on core/controllers/access_validators.py:
```diff
@@ -110,8 +116,15 @@ class ProfileExistsValidationHandler(
+        Raises:
+            PageNotFoundException. No user settings found for the given
+                username.
+        """
```
Here is the 'Raises'.

deadex-ng commented on core/controllers/access_validators.py:
```diff
@@ -221,6 +243,11 @@ class BlogPostPageAccessValidationHandler(
+        Raises:
+            PageNotFoundException. The blog post cannot be found.
+        """
         assert self.normalized_request is not None
```
Here is 'Raises'.

deadex-ng commented on core/controllers/admin.py:
```diff
@@ -1331,6 +1384,12 @@ class TopicManagerRoleHandler(
+        Raises:
+            InvalidInputException. User with given username does not exist.
+        """
         assert self.normalized_payload is not None
```
Here is 'Raises'.

deadex-ng commented on core/controllers/blog_dashboard.py:
```diff
@@ -325,7 +336,14 @@ def put(self, blog_post_id: str) -> None:
+        Raises:
+            InvalidInputException. The input provided is not valid.
+        """
         assert self.normalized_request is not None
```
Here is 'Raises'.

deadex-ng commented on core/controllers/blog_dashboard.py:
```diff
@@ -349,7 +367,11 @@ def post(self, blog_post_id: str) -> None:
+        Args:
+            blog_post_id: str. The ID of the blog post.
+        """
         blog_services.delete_blog_post(blog_post_id)
```
Updated this to 'Deletes a blog post.'

deadex-ng commented on core/controllers/blog_homepage.py:
```diff
@@ -249,7 +249,15 @@ class BlogPostDataHandler(
+            PageNotFoundException. The blog post page with the given url
+                doesn't exist.
+        """
         blog_post = blog_services.get_blog_post_by_url_fragment(blog_post_url)
```
Here is 'Raises'.

deadex-ng commented on core/controllers/classroom.py:
```diff
@@ -245,7 +249,15 @@ class ClassroomHandler(
+            PageNotFoundException. The classroom with the given id or
+                url doesn't exist.
+        """
         classroom = classroom_config_services.get_classroom_by_id(
```
Here is 'Raises'.

deadex-ng commented:
@JayVivarekar PTAL. Thanks, I have left comments on the sections you said.

JayVivarekar reviewed (approved):
Thanks @deadex-ng! LGTM for codeowners

seanlip commented:
I am a bit confused about this PR. Is @lkbhitesh07 or is @deadex-ng supposed to be assigned to it?

deadex-ng commented:
> I am a bit confused about this PR. Is @lkbhitesh07 or is @deadex-ng supposed to be assigned to it?

@seanlip I'm waiting for a review from @lkbhitesh07, he requested some changes and I addressed them [here.](https://github.com/oppia/oppia/pull/18216#pullrequestreview-1443230962)

seanlip commented:
OK, thanks @deadex-ng. Please follow the instruction in the second bullet point under "PR Pointers" in the first comment to assign him, otherwise he won't see this PR.  In general let's keep the assignees field up to date.

seanlip commented:
Btw @deadex-ng your backend tests are failing.

deadex-ng commented:
> Hey @deadex-ng, thanks for the changes. I have added some comments. The general comment for the `admin.py` file would be to avoid the usage of `Handles __ requests` part and to add the `Raises` part where missing. Wherever I mention "ditto" I mean these 2 changes mainly. Please assign me again after the changes.
> 
> Thanks

@lkbhitesh07 PTAL

deadex-ng commented:
I'm waiting for your review @lkbhitesh07 PTAL.

Thanks.

deadex-ng committed:
Fix failing backend test

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1178,6 +1209,17 @@ class AdminRoleHandler(
     def get(self) -> None:
+        """Retrieves information about users based on different filter
+        criteria. It expects a request of type
+        AdminRoleHandlerNormalizedGetRequestDict
```
```suggestion
        """Retrieves information about users based on different filter
        criteria to populate the roles tab.
```

deadex-ng replied:
I have updated the docstring.

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1416,6 +1475,11 @@ class BannedUsersHandler(
 
     @acl_decorators.can_access_admin_page
     def put(self) -> None:
+        """Marks a user as banned by their username.
```
```suggestion
        """Marks a user as banned.
```

deadex-ng replied:
I have updated the docstring.

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1430,6 +1494,11 @@ def put(self) -> None:
 
     @acl_decorators.can_access_admin_page
     def delete(self) -> None:
+        """Removes the banned status of user.
```
```suggestion
        """Removes the banned status of the user.
```

deadex-ng replied:
I have updated the docstring.

lkbhitesh07 commented on core/controllers/admin.py:
```diff
@@ -1687,6 +1787,16 @@ class UpdateUsernameHandler(
 
     @acl_decorators.can_access_admin_page
     def put(self) -> None:
+        """Updates the username, profile picture, and logs the username
+        change for a user.
```
```suggestion
        """Updates the username for a user.
```

deadex-ng replied:
I have updated the docstring.

lkbhitesh07 reviewed (changes requested):
Hey @deadex-ng, thanks for the changes. I have left some comments, PTAL. Please assign me after the changes are done.
Thanks

lkbhitesh07 commented:
Also please note that some of the tests are failing.

deadex-ng committed:
Added/updated args, return and raises in docstring

deadex-ng commented:
> Also please note that some of the tests are failing.

Yes, I'm trying to figure out why some tests are failing.

deadex-ng commented:
I have made the changes, @lkbhitesh07 PTAL.

lkbhitesh07 reviewed (approved):
Thanks @deadex-ng for the changes. Looks good to me for the codeowners files.
Thanks

deadex-ng commented:
> Also please note that some of the tests are failing.

@lkbhitesh07 I have looked into it and opened discussion #18348, I haven't figured out why the tests are failing.

deadex-ng committed:
Merge remote-tracking branch 'upstream/develop' into fix_11776

lkbhitesh07 commented on core/controllers/practice_sessions.py:
```diff
@@ -47,8 +47,8 @@ class PracticeSessionsPage(
     @acl_decorators.can_access_topic_viewer_page
-    def get(self, _: str) -> None:
-        """Handles GET requests."""
+    def get(self) -> None:
+        """Renders the practice session page."""
```
Hey @deadex-ng, I think this is where the backend test is failing. Note that you have made changes to the `get` function where you have removed `, _: str`, restore it back and I think then the tests will pass.
Thanks

deadex-ng replied:
@lkbhitesh07 Yes, you're right. The I have restored it and the backend tests have passed.
Thanks.

deadex-ng committed:
Restore function Args

deadex-ng committed:
Merge remote-tracking branch 'upstream/develop' into fix_11776

deadex-ng commented:
All tests have passed and I guess it's ready to merge,  @seanlip PTAL.

seanlip commented:
@deadex-ng Done. Congrats on your second PR! :)

Once this is merged, feel free to follow the instructions in step 8 [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up) to be added as a collaborator. Thanks for the PR!

deadex-ng commented:
> @deadex-ng Done. Congrats on your second PR! :)
> 
> Once this is merged, feel free to follow the instructions in step 8 [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#setting-things-up) to be added as a collaborator. Thanks for the PR!

I have filled out the form and thanks @seanlip.
